### PR TITLE
Add `DepositModalResolver` - closes #1668

### LINKED ISSUE #1668: Add DepositModalResolver component
A component that renders a DepositModal component based on the url params `chain`, `backingToken` and `protocol`

### PR DESCRIPTION
**Pull request by danieledamiani**

### CODE REVIEW

**Comment by lazarmitic** [comment]
in review

File: packages/tempus-client_v3/src/components/DepositModal/DepositModalResolver.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  const [startDate] = useState<Date>(new Date());
+  const [maturityTerms] = useState<MaturityTerm[]>([{ apr: new Decimal('1'), date: new Date() }]);
+  const singleCurrencyUsdRates = new Map<Ticker, Decimal>();
+
+  console.log('modalResolver ===>', ticker, protocol, chain);
+
+  singleCurrencyUsdRates.set('ETH', new Decimal(3500));
+
+  const handleCloseModal = () => {
+    navigate('/');
```
Comment (lazarmitic) [inline packages/tempus-client_v3/src/components/DepositModal/DepositModalResolver.tsx:22]: If user opens deposit modal from Portfolio page and closes it, we want to go back to portfolio page, not the markets page.

I think we can just navigate one step back here (just like when user clicks back button in the browser).

Comment (danieledamiani) [reply packages/tempus-client_v3/src/components/DepositModal/DepositModalResolver.tsx:22]: good idea

Comment (danieledamiani) [reply packages/tempus-client_v3/src/components/DepositModal/DepositModalResolver.tsx]: changed

**Comment by ShawTim** [comment]
reviewing

File: packages/tempus-client_v3/src/components/DepositModal/DepositModalResolver.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  const { chain, ticker, protocol } = useParams();
+
+  const [precision] = useState<number>(DEFAULT_TOKEN_PRECISION);
+  const [startDate] = useState<Date>(new Date());
+  const [maturityTerms] = useState<MaturityTerm[]>([{ apr: new Decimal('1'), date: new Date() }]);
+  const singleCurrencyUsdRates = new Map<Ticker, Decimal>();
+
+  console.log('modalResolver ===>', ticker, protocol, chain);
+
+  singleCurrencyUsdRates.set('ETH', new Decimal(3500));
```
Comment (ShawTim) [inline packages/tempus-client_v3/src/components/DepositModal/DepositModalResolver.tsx:19]: we have `useTokenRates` to get the actual rate here. or, we may want to add `// TODO` here to remind we still have sth to integrate

**Review by lazarmitic — APPROVED** [review]

File: packages/tempus-client_v3/src/components/DepositModal/DepositModalResolver.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  const handleCloseModal = () => {
```
Comment (iamvukasin) [inline packages/tempus-client_v3/src/components/DepositModal/DepositModalResolver.tsx:21]: Do we need `useCallback` here?

File: packages/tempus-client_v3/src/mocks/config/mockConfig.ts [hunk 4]
```diff
@@ -70,8 +76,10 @@ export const pool4 = {
   chain: 'fantom',
   protocol: 'yearn',
   protocolDisplayName: 'Yearn',
+  spotPrice: '0.0002',
+  decimalsForUI: 4,
   startDate: 1647216000000, // 2022/03/14
-  maturityDate: new Date(2025, 0, 1).getTime(),
+  maturityDate: 1671062400000, // 2022/12/15
```
Comment (ShawTim) [inline packages/tempus-client_v3/src/mocks/config/mockConfig.ts:82]: do we want to calculate the `maturityDate` here? such that no matter when we run the test we can ensure we have X matured pools and Y immatured pools which will affect the test results of some test cases

File: packages/tempus-client_v3/src/components/DepositModal/DepositModalResolver.tsx (hunk 3 above)
Comment (danieledamiani) [reply packages/tempus-client_v3/src/components/DepositModal/DepositModalResolver.tsx:21]: I may add it during refactor

File: packages/tempus-client_v3/src/mocks/config/mockConfig.ts (hunk 4 above)
Comment (danieledamiani) [reply packages/tempus-client_v3/src/mocks/config/mockConfig.ts:82]: `new Date(2025, 0, 1).getTime() === 1671062400000`, I just removed some unnecessary calculation

Comment (ShawTim) [reply packages/tempus-client_v3/src/mocks/config/mockConfig.ts:82]: ya i mean the time should depend on the current time, e.g. 6 months from the current time, such that our test can get some expected results.
may not be related to this PR though. we can improve it in another PR

**Review by ShawTim — APPROVED** [review]

**Pull request merged**